src/frontend/src/planner/select.rs

```diff
@@ -413,6 +428,139 @@ impl Planner {
         Ok((root, exprs))
     }
 
+    /// Substitutes all [`Subquery`] in `exprs`.
+    ///
+    /// Each time a [`Subquery`] is found, it is replaced by a new [`InputRef`]. And `root` is
+    /// replaced by a new `LeftOuter` [`LogicalApply`] whose left side is `root` and right side is
+    /// the planned subquery.
+    ///
+    /// The [`InputRef`]s' indexes start from `root.schema().len()`,
+    /// which means they are additional columns beyond the original `root`.
+    ///
+    ///
+    /// The cross join way only meaningful when there are multiple subqueries
+    ///
+    ///            Apply
+    ///           /    \
+    ///         left   CrossJoin
+    ///                   /   \
+    ///           Subquery1   CrossJoin
+    ///                         /   \
+    ///                 Subquery2   Subquery3
```

Comment:
```
    ///            Apply
    ///           /    \
    ///         left   CrossJoin
    ///                   /   \
    ///           CrossJoin  Subquery3
    ///              /   \
    ///    Subquery1   Subquery2
```
Looks like the CrossJoin part is left-deep, according to the implementation below
https://github.com/risingwavelabs/risingwave/blob/1d461bd1a039994836d7c944a84ab0e4a77b13d1/src/frontend/src/planner/select.rs#L529-L538